raft-tech/cfgov-refresh
# Jb narrative modals

This PR will add narrative notifications to the application to guide the user through certain CFPB approved screens on the users first visit.


## Changes

- Added flags to localStorage session to track initial visit to the calendar view, after first expense or income entry, first visit to SNAP benefits form, and first visit to Fix It strategies.


## How to test this PR

1. Pull down this branch, ```jb-narrative-modals```
2. Start server, run app locally
3. Enter initial income (going through wizard)
- expect to see first modal on initial arrival to calendar view.
- expect to see second modal after entering first income/expense
- expect to see modal on first visit to SNAP benefit form
- expect to see modal on first visit to Fix It Strategies page after clicking on red Fix It button if week ending balance is negative.



## Screenshots
#### Initial welcome
<img width="762" alt="Screen Shot 2020-06-18 at 2 30 17 PM" src="https://user-images.githubusercontent.com/21065244/85068907-6eb48800-b170-11ea-8e39-686f6c10e211.png">

#### After entering first expense/income
<img width="754" alt="Screen Shot 2020-06-18 at 2 30 46 PM" src="https://user-images.githubusercontent.com/21065244/85068916-71af7880-b170-11ea-8275-4a2c14395540.png">

#### First visit to SNAP
<img width="757" alt="Screen Shot 2020-06-18 at 2 30 55 PM" src="https://user-images.githubusercontent.com/21065244/85068923-74aa6900-b170-11ea-9442-404aaea74d66.png">

#### First visit to FixIt Page
<img width="778" alt="Screen Shot 2020-06-18 at 2 31 24 PM" src="https://user-images.githubusercontent.com/21065244/85068935-770cc300-b170-11ea-9e27-db483fe80e74.png">
